src/MatrixClientPeg.ts

```diff
@@ -273,15 +274,20 @@ class MatrixClientPegClass implements IMatrixClientPeg {
         opts.threadSupport = true;
 
         if (SettingsStore.getValue("feature_sliding_sync")) {
+            const baseUrl = this.matrixClient.baseUrl;
             const proxyUrl = SettingsStore.getValue("feature_sliding_sync_proxy_url");
+            const wellKnown = AutoDiscovery.findClientConfig(baseUrl); // the client isn't init'd yet
+            const wellKnownProxyUrl = (await wellKnown)?.["org.matrix.msc3575.proxy"]?.url;
             if (proxyUrl) {
-                logger.log("Activating sliding sync using proxy at ", proxyUrl);
+                logger.log("Activating sliding sync using manually added proxy at ", proxyUrl);
+            } else if (wellKnownProxyUrl) {
+                logger.log("Activating sliding sync using well-known proxy at ", wellKnownProxyUrl);
             } else {
-                logger.log("Activating sliding sync");
+                logger.log("Activating sliding sync using the HS base url at ", baseUrl);
             }
             opts.slidingSync = SlidingSyncManager.instance.configure(
                 this.matrixClient,
-                proxyUrl || this.matrixClient.baseUrl,
+                proxyUrl || wellKnownProxyUrl || baseUrl,
             );
             SlidingSyncManager.instance.startSpidering(100, 50); // 100 rooms at a time, 50ms apart
         }
```

Comment:
Could we extract this logic to somewhere else just for the sake of cleanliness, maybe `SlidingSyncManager.instance.setup()` or something